skimage/filters/thresholding.py

```diff
@@ -959,7 +966,9 @@ def threshold_triangle(image, nbins=256):
 
     if arg_low_level == arg_high_level:
         # Image has constant intensity.
-        return image.ravel()[0]
+        if image is not None:
+            return image.ravel()[0]
+        return bin_centers[arg_low_level]
```

Comment:
If we use
```suggestion
        return bin_centers[arg_low_level]
```
we can simplify the logic a bit?

Reply:
I initially tried this, but then got test failures for `float` type images:

```
FAILED scikit-image/filters/tests/test_thresholding.py::test_triangle_uniform_images[float16] - AssertionError: assert 0.001953 == 0
FAILED scikit-image/filters/tests/test_thresholding.py::test_triangle_uniform_images[float32] - AssertionError: assert 0.001953125 == 0
```
(see https://github.com/scikit-image/scikit-image/actions/runs/14352903789/job/40235602154#step:9:555)

.. so I decided to keep existing functionality unchanged where possible.